### PR TITLE
Fix: `select-notifications` causes button overlap on mobile

### LINKED ISSUE #5796: `select-notifications` z-index bug on mobile
### Please ensure:

- [X] The bug is caused by Refined GitHub. It doesn't happen if I disable the extension.
- [X] The bug happens after clearing extension cache. The "Clear cache" button below can also be found at the bottom of the options page.

### Example URLs

Notifications page 

### Extension version

22.7.1

### Browser(s) used

Safari iOS 

### Description

We push these buttons above the overlay so that they’re clickable when the dropdown is open. This is no longer feasible when the dropdown turns into a modal. 

We should move the z-index change to a media query that matches this UI change. 

B563F813-F391-4E1A-B061-CCF9BD33DBCE

### Include in this issue:

- [X] Screenshots/video/gif demonstrating the bug, if it’s visual
- [ ] Console errors, if any

### PR DESCRIPTION
**Pull request by fregante**
- Fix #5796

## Test URLs

https://github.com/notifications

## Before

Screen Shot

## After

Screen Shot 1

### CODE REVIEW

**Comment by fregante** [comment]
I forgot to test whether this still works where it should though 🥲

File: source/features/select-notifications.css [hunk 1]
```diff
@@ -1,5 +1,9 @@
-.js-notifications-mark-selected-actions > *,
-.rgh-open-selected-button {
-	/* Allow clicking on "Done" and other toolbar buttons even when the dropdown (z-index of 99) is open #4753 */
-	z-index: 100;
+/* Only when the meny is shown as a dropdown, not as a modal (mobile) #5796 */
```
Comment (yakov116) [inline source/features/select-notifications.css:1]: Meny?

Comment (fregante) [reply source/features/select-notifications.css:1]: https://meny.no 😂

Comment (fregante) [reply source/features/select-notifications.css:1]: When you think _it's just a one-line change_ but it has 5 bugs.

**Pull request merged**